app/models/ems_refresh/parsers/kubernetes.rb

```diff
@@ -121,9 +121,13 @@ def parse_node(node)
         parse_node_condition(condition)
       end
 
-      # Searching for the underlying instance. At the moment this search strategy
-      # is tested only with OpenStack.
-      vms = VmOpenstack.where(:uid_ems => new_result[:identity_system].downcase)
+      # Establish a relationship between this node and the vm its on (if its in the system)
+      # supported relationships: oVirt and Openstack
+      types = [ManageIQ::Providers::Redhat::InfraManager::Vm.name,
+               VmOpenstack.name]
+
+      # Searching for the underlying instance for Openstack or oVirt.
+      vms = Vm.where(:uid_ems => new_result[:identity_system].downcase, :type => types)
```

Comment:
ActiveRecord STI creates the first type check.  I'm surprised it generates one though when type is given in a where clause.   Technically the query engine should detect that one is a subset of the other and basically ignore the first one.  You might be able to verify this with EXPLAIN.

Still wonder if this is an issue with ActiveRecord though.  @tenderlove @matthewd ?